### ISSUE TITLE
[RDataFrame] RDataFrame.Redefine does not work with Snapshot

### ISSUE THREAD

**Issue by AlkaidCheng** [root-project/root#10233]
After calling the RDataFrame.Redefine method, saving the Snapshot containing the redefined column will raise an error. 

Reproducer:
```
import ROOT
import numpy as np
rdf = ROOT.RDF.MakeNumpyDataFrame({"bar": np.arange(0, 10, 1)})
rdf = rdf.Redefine("bar", "bar + 1")
rdf.Snapshot("output", "output.root")
```

This gives:
```
TypeError: Template method resolution failed:
  none of the 3 overloaded methods succeeded. Full details:
  ROOT::RDF::RResultPtr<ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void> > ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void>::Snapshot(basic_string_view<char,char_traits<char> > treename, basic_string_view<char,char_traits<char> > filename, initializer_list<string> columnList, const ROOT::RDF::RSnapshotOptions& options = ROOT::RDF::RSnapshotOptions()) =>
    TypeError: takes at least 3 arguments (2 given)
  ROOT::RDF::RResultPtr<ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void> > ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void>::Snapshot(basic_string_view<char,char_traits<char> > treename, basic_string_view<char,char_traits<char> > filename, const vector<string>& columnList, const ROOT::RDF::RSnapshotOptions& options = ROOT::RDF::RSnapshotOptions()) =>
    TypeError: takes at least 3 arguments (2 given)
  ROOT::RDF::RResultPtr<ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void> > ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void>::Snapshot(basic_string_view<char,char_traits<char> > treename, basic_string_view<char,char_traits<char> > filename, basic_string_view<char,char_traits<char> > columnNameRegexp = "", const ROOT::RDF::RSnapshotOptions& options = ROOT::RDF::RSnapshotOptions()) =>
    logic_error: Error: column "bar" was passed to Snapshot twice. This is not supported: only one of the columns would be readable with RDataFrame.
  ROOT::RDF::RResultPtr<ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void> > ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void>::Snapshot(basic_string_view<char,char_traits<char> > treename, basic_string_view<char,char_traits<char> > filename, basic_string_view<char,char_traits<char> > columnNameRegexp = "", const ROOT::RDF::RSnapshotOptions& options = ROOT::RDF::RSnapshotOptions()) =>
    logic_error: Error: column "bar" was passed to Snapshot twice. This is not supported: only one of the columns would be readable with RDataFrame.
```

However, this works:
```
import ROOT
import numpy as np
rdf = ROOT.RDF.MakeNumpyDataFrame({"bar": np.arange(0, 10, 1)})
rdf = rdf.Redefine("bar", "bar + 1")
rdf.Snapshot("output", "output.root", ["bar"])
```

ROOT Version: 6.26/00 (conda install), 6.27/01 (swan bleeding edge)

**Comment by eguiraud** [comment]
There are two issues:

1. Snapshot does not de-duplicate columns from data-sources and columns from Redefines when deciding which columns to write out
2. the logic in Snapshot that creates output branches assumes that if there is an input branch for column "x", then that _must_ be what we want to write out (which was the case until `Redefine` was introduced)  

1 is easy to fix, I'm now looking into fixing 2. We definitely want these fixes in 6.26.02.

**Comment by eguiraud** [comment]
This issue should be fixed by https://github.com/root-project/root/pull/10319

**Comment by eguiraud** [comment]
Thank you again for reporting @AlkaidCheng . This is fixed in the master branch (which you can test out via the [nightly builds](https://root.cern/install/nightlies/)) and the fix will be available in the upcoming v6.26.02 patch release. Let us know if you encounter any further issues.

**Comment by AlkaidCheng** [comment]
Thanks a lot for the follow-up!
